### ISSUE TITLE
selection file conversion failed

### ISSUE THREAD

**Issue by AtefBN** [ESGF/esgf-download#2]
Running any esgpull convert for this selection file ends with an error: 

`
(esgpull) -bash-4.2$ esgpull convert 2023-02-09_*******.txt --graph

2023-04-11 10:51:48]  ERROR     root

Traceback (most recent call last):
  File "/gpfscmip/gpfsdata/esgf/miniconda/envs/esgpull/lib/python3.11/site-packages/esgpull/tui.py", line 154, in logging
    yield
  File "/gpfscmip/gpfsdata/esgf/miniconda/envs/esgpull/lib/python3.11/site-packages/esgpull/cli/convert.py", line 237, in convert
    graph = convert_file(path)
            ^^^^^^^^^^^^^^^^^^
  File "/gpfscmip/gpfsdata/esgf/miniconda/envs/esgpull/lib/python3.11/site-packages/esgpull/cli/convert.py", line 121, in convert_file
    query.tags = result.name[0]
    ^^^^^^^^^^
  File "/gpfscmip/gpfsdata/esgf/miniconda/envs/esgpull/lib/python3.11/site-packages/sqlalchemy/orm/attributes.py", line 528, in __set__
    self.impl.set(
  File "/gpfscmip/gpfsdata/esgf/miniconda/envs/esgpull/lib/python3.11/site-packages/sqlalchemy/orm/attributes.py", line 1919, in set
    raise TypeError(
TypeError: Incompatible collection type: str is not list-like
`
to replicate selection file in question :

mip_era=CMIP6
source_id=CESM2 CMCC-ESM2 CanESM5-CanOE GISS-E2-1-G UKESM1-0-LL
experiment_id=piControl
#variant_label=r1i1p1f1
frequency=mon
variable_id=phydiat mlotstmax tos si no3 dfe vo psl

**Comment by svenrdz** [comment]
It seems that this file (filename = `areacella_fx_FGOALS-g2_lgm_r0i0p0.nc`) has no `checksum` nor `checksum_type` attributes in the synda CMIP5 database you are importing, and those are currently required by esgpull.
The query ran by esgpull on the ESGF search API confirms that the 2 attributes are also missing from the file's metadata: https://esgf-node.ipsl.upmc.fr/esg-search/search?type=File&offset=0&limit=1&format=application%2Fsolr%2Bjson&fields=%2A&query=title%3Aareacella_fx_FGOALS-g2_lgm_r0i0p0.nc&distrib=true&latest=true&retracted=false
I am guessing synda used the same query to fill its database at the time this file was added.
Now if I increase the `limit` parameter for this query (`numFound` tells us 4 replicas exist in this case), `checksum` and `checksum_type` do exist in the next 3 replicas' metadata.

Knowing this, 2 things could be done during import to handle missing information:
- look up the metadata from **all** replicas for each incomplete file and use the most complete one, but I don't know if we can guarantee there will always be at least one index node with the full metadata,
- skip files with incomplete metadata from the imported database

The first solution might look more complete but it could seriously slow down the import procedure, and does not guarantee missing info will be filled, while the 2nd solution is easy to set up but will definitely introduce divergence between the filesystem and database.

**Comment by svenrdz** [comment]
The issue cannot be replicated with the selection you provided since it comes from the first line (commented email address I suppose) which is used as a tag for the query produced with the conversion.
It comes from a regression introduced with a change on how `Query` instances are created (9acf164f88a31476f5f87b8847c7f1cd46f7d3b0).
The issue is now fixed with 3d9ae0d3665f89b1e703a9b6af996e7b82f00856 and I bumped the version to `0.3.8`.
